testing/src/main/java/io/stargate/it/http/graphql/cqlfirst/GraphqlApolloTest.java

```diff
@@ -2100,6 +2102,232 @@ public void udtsTest() {
+            .description("d2")
+            .build();
+
+    ApolloClient client = getApolloClient("/graphql/betterbotz");
+    InsertOrdersMutation order1Mutation = InsertOrdersMutation.builder().value(order1).build();
+
+    InsertOrdersMutation order2Mutation = InsertOrdersMutation.builder().value(order2).build();
+
+    assertThat(getObservable(client.mutate(order1Mutation)).getInsertOrders().isPresent()).isTrue();
+    assertThat(getObservable(client.mutate(order2Mutation)).getInsertOrders().isPresent()).isTrue();
+
+    OrdersFilterInput filterInput =
+        OrdersFilterInput.builder().prodName(StringFilterInput.builder().eq("p1").build()).build();
+    GetOrdersWithSumQuery query = GetOrdersWithSumQuery.builder().filter(filterInput).build();
+
+    GetOrdersWithSumQuery.Data result = getObservable(client.query(query));
+
+    assertThat(result.getOrders().get().getValues().get().get(0).getSum().get()).isEqualTo("5500");
+  }
+
+  @Test
+  @DisplayName("Should test all available function types")
+  public void testAllGraphqlFunctionTypes() {
+
+    Number value = 100;
+    OrdersInput order1 =
+        OrdersInput.builder()
+            .prodName("p1")
+            .customerName("c1")
+            .price(value.toString())
+            .value_int(value.intValue())
+            .value_double(value.doubleValue())
+            .value_bigint(value.toString())
+            .value_varint(value.toString())
+            .value_float(value.doubleValue())
+            .value_smallint(value.shortValue())
+            .value_tinyint(value.byteValue())
+            .description("d1")
+            .build();
+
+    ApolloClient client = getApolloClient("/graphql/betterbotz");
+    InsertOrdersMutation order1Mutation = InsertOrdersMutation.builder().value(order1).build();
+
+    InsertOrdersMutation.Data result = getObservable(client.mutate(order1Mutation));
+    assertThat(result.getInsertOrders().isPresent()).isTrue();
+
+    OrdersFilterInput filterInput =
+        OrdersFilterInput.builder().prodName(StringFilterInput.builder().eq("p1").build()).build();
+
+    GetOrdersAllFunctionsQuery query =
+        GetOrdersAllFunctionsQuery.builder().filter(filterInput).build();
+
+    GetOrdersAllFunctionsQuery.Data functionsResult = getObservable(client.query(query));
+
+    GetOrdersAllFunctionsQuery.Value res =
+        functionsResult.getOrders().get().getValues().get().get(0);
+    assertThat(res.get_int_function().get()).isEqualTo(value.intValue());
+    assertThat(res.get_double_function().get()).isEqualTo(value.doubleValue());
+    assertThat(res.get_bigint_function().get()).isEqualTo(value.toString());
+    assertThat(res.get_decimal_function().get()).isEqualTo(value.toString());
+    assertThat(res.get_varint_function().get()).isEqualTo(value.toString());
+    assertThat(res.get_float_function().get()).isEqualTo(value.doubleValue());
+    assertThat(res.get_smallint_function().get()).isEqualTo(BigDecimal.valueOf(value.shortValue()));
```

Comment:
the byte and short values are returned as `BigDecimal`:
```
smallint_function = {Optional@7638} "Optional[100]"
 value = {BigDecimal@7648} "100"
_tinyint_function = {Optional@7639} "Optional[100]"
 value = {BigDecimal@7650} "100"
```
It seems suspicious. @olim7t is it intended or do we have bugs in both `smallint` and `tinyint` GraphQl conversions?

Reply:
This might be specific to the Apollo client library. These types map to custom scalars. Apollo can't guess what the actual range is, so I imagine when it sees a number it picks the most general representation.
There are ways to register [custom adapters](https://www.apollographql.com/docs/android/essentials/custom-scalar-types/) on the client, but I'm not sure if it's worth it for these tests (especially given that I think they shouldn't use Apollo).

Reply:
yeah, that was it. Resolving